test/e2e/twofa_test.go

```diff
@@ -0,0 +1,118 @@
+package hardwaretest
+
+import (
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"strings"
+	"testing"
+
+	root "github.com/equinix/metal-cli/internal/cli"
+	outputPkg "github.com/equinix/metal-cli/internal/outputs"
+	"github.com/equinix/metal-cli/internal/twofa"
+	"github.com/spf13/cobra"
+)
+
+var mockOtpUri = "otpauth://totp/foo"
+
+func setupMock() *root.Client {
+	mockAPI := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		var responseBody string
+		if r.URL.Path == "/user/otp/sms/receive" {
+			w.WriteHeader(http.StatusNoContent)
+		} else if r.URL.Path == "/user/otp/app/receive" {
+			w.Header().Add("Content-Type", "application/json")
+			responseBody = fmt.Sprintf(`{"otp_uri": "%v"}`, mockOtpUri)
+
+		} else {
+			responseBody = fmt.Sprintf("no mock for endpoint %v", r.URL.Path)
+			w.WriteHeader(http.StatusNotImplemented)
+		}
+		_, err := w.Write([]byte(responseBody))
+		if err != nil {
+			log.Fatalf("Failed to write mock response: %v", err)
+		}
+	}))
+	mockClient := root.NewClient("", mockAPI.URL, "metal")
+	return mockClient
+
+}
+
+func TestCli_Twofa(t *testing.T) {
+	subCommand := "2fa"
+	// Adjust this response as needed for your tests.
+
+	rootClient := setupMock()
+
+	type fields struct {
+		MainCmd  *cobra.Command
+		Outputer outputPkg.Outputer
+	}
+	tests := []struct {
+		name    string
+		fields  fields
+		want    *cobra.Command
+		cmdFunc func(*testing.T, *cobra.Command)
+	}{
+		{
+			name: "receive sms",
```

Comment:
**Note:** this is only testing the receive subcommand because I haven't recreated a successful scenario with the `enable` and `disable` commands.  For the other 2fa subcommands, the same failure happens in both the packngo and equinix-sdk-go implementations:

```
HTTP/1.1 424 Failed Dependency
{"errors":["Invalid OTP"]}
```

Since this is a pre-existing issue and isn't caused or fixed by the SDK migration, I'm advocating to handle it in a separate issue & PR.